subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/platforms/JavaPlatformResolveIntegrationTest.groovy

```diff
@@ -263,6 +263,49 @@ class JavaPlatformResolveIntegrationTest extends AbstractHttpDependencyResolutio
         }
     }
 
+    def "can resolve a platform with a constraint to determine the platform version"() {
+        def platform = mavenHttpRepo.module("org", "platform", "1.0")
+                .hasType("pom")
+                .allowAll()
+                .publish()
+
+        when:
+        buildFile << """
+            dependencies {
+                api platform("org:platform") // no version, will select the "platform" component
+                constraints {
+                   api platform("org:platform:1.0")
```

Comment:
❌ This version of the test will pass with or without your changes.
There needs to be a version that tries with `api "org:platform:1.0"` as the constraint.